capture/crawler_valor.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+from goose import Goose
+import pymongo
+from bs4 import BeautifulSoup
+import requests
+import re
+import pandas as pd
+import datetime
+import zlib
+import cPickle as CP
+import cld
+import sys
+from requests.exceptions import ConnectionError, MissingSchema, Timeout
+import bson
+import settings
+import logging_mc
+
+logger = logging_mc.get_logger( 'valor' )
+
+
+client = pymongo.MongoClient(settings.MONGOHOST, 27017)
+MCDB = client.MCDB
+ARTICLES = MCDB.articles  # Article Collection
+ARTICLES.ensure_index("source")
+
+def find_articles():
+	"""
+	Get the urls of last news
+	:return: last news' urls of all categories
+	:rtype: set()
+	"""
+	urls = ['http://www.valor.com.br/ultimas-noticias/brasil',
+			'http://www.valor.com.br/ultimas-noticias/politica',
+			'http://www.valor.com.br/ultimas-noticias/financas',
+			'http://www.valor.com.br/ultimas-noticias/empresas',
+			'http://www.valor.com.br/ultimas-noticias/agro',
+			'http://www.valor.com.br/ultimas-noticias/internacional',
+			'http://www.valor.com.br/ultimas-noticias/opiniao',
+			'http://www.valor.com.br/ultimas-noticias/legislacao',
+			'http://www.valor.com.br/ultimas-noticias/carreira',
+			'http://www.valor.com.br/ultimas-noticias/cultura']
+	news_urls = list()
+	for INDEX_URL in urls:
+		index = requests.get(INDEX_URL).content
+		soup = BeautifulSoup(index, "lxml")
+		news_index = soup.find(id="block-valor_capa_automatica-central_automatico").find_all('h2')
+		news_urls = news_urls + ['http://www.valor.com.br' + BeautifulSoup(  art.encode('utf8') , "lxml" ).find('a').attrs['href'] for art in news_index]
```

Comment:
It might be a good idea to break this line to make it more readable. PEP8 suggests 72 with a maximum of 79. I like to follow that whenever possible.